.github/workflows/bindings_ci.yml

```diff
@@ -116,3 +116,46 @@ jobs:
     with:
         use_rust_sdk: "." # use local checkout
         use_complement_crypto: "MATCHING_BRANCH"
+
+  test-crypto-apple-framework-generation:
+    name: Generate Crypto FFI Apple XCFramework
+    runs-on: macos-12
+    if: github.event_name == 'push' || !github.event.pull_request.draft
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # install protoc in case we end up rebuilding opentelemetry-proto
+      - name: Install protoc
+        uses: taiki-e/install-action@v2
+        with:
+          tool: protoc@3.20.3
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Add rust targets
+        run: |
+          rustup target add aarch64-apple-ios
+          rustup target add aarch64-apple-darwin
+          rustup target add x86_64-apple-darwin
+          rustup target add aarch64-apple-ios-sim
+          rustup target add x86_64-apple-ios
+
+      # Cargo config can screw with caching and is only used for alias config
+      # and extra lints, which we don't care about here
+      - name: Delete cargo config
+        run: rm .cargo/config.toml
+
+      - name: Load cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          save-if: ${{ github.ref == 'refs/heads/main' }}
+
+      - name: Run the Build Framework script
+        run: ./bindings/apple/build_crypto_xcframework.sh -i
```

Comment:
The `-i` option describes the fact you want to test for iOS only. Then, this is not necessary to add the following targets with `rustup` hereinabove (in the “Add rust targets” step):
- `aarch64-apple-darwin`
- `x86_64-apple-darwin`
- `aarch64-apple-ios-sim`
- `x86_64-apple-ios`.

Only the `aarch64-apple-ios` target can be kept, according to https://github.com/matrix-org/matrix-rust-sdk/pull/3334/files#diff-3bbe5c410acf8c549fb356c29229df7b45a37e8db0932b773ce6294cc3935262R44.

Reply:
Done https://github.com/matrix-org/matrix-rust-sdk/pull/3334/commits/b430d95c0aab174a471577f262e91e5a38defa77